src/Microsoft.ML.ImageAnalytics/ImageBase.cs

```diff
@@ -0,0 +1,480 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using SkiaSharp;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+
+namespace Microsoft.ML.Data
+{
+    /// <summary>
+    /// Provides the base class for the image provider which allow registering a provider to use instead of the default provider.
+    /// </summary>
+    internal abstract class ImageProvider
```

Comment:
@michaelgsharp @ericstj @luisquintanilla 
I have made this internal for now, but we can make it public if you think it is better to expose it. These APIs will allow users to provide their own imaging handling (to use other imaging library) and not necessarily depend on our internal implementation. I am fine either way.

Reply:
I think eventually it would be good to do, but I think initially having them internal would be good incase we need to change anything we find after using them.

Reply:
Yes, we scoped the exposure to only what ML.NET needs and not support extensibility. the last commit in this PR showing the change.